netjsonconfig/backends/airos/airos.py

```diff
@@ -0,0 +1,166 @@
+        for k, v in self.intermediate_data.items():
+            self.intermediate_data[k] = [x for x in flatten(intermediate_to_list(v)) if x != {}]
+
+
+def flatten(xs):
+    """
+    Flatten a list
+    """
+    if xs is not list:
+        return xs
+    else:
+        return reduce(lambda x, y: x + flatten(y), xs, [])
+
+
+def intermediate_to_list(configuration):
+    """
+    Explore the configuration tree and flatten where
+    possible with the following policy
+    - list -> prepend the list index to every item key
+    - dictionary -> prepend the father key to every key
+
+    configuration :: List[Enum[Dict,List]]
+    return List[Dict]
+
+    >>> intermediate_to_list([
+        {
+            'spam': {
+                'eggs': 'spam and eggs'
+            }
+        }
+    ])
+    >>>
+    [{
+            'spam.eggs' : 'spam and eggs'
+    ]}
+
+    >>> intermediate_to_list([
+        {
+            'spam': {
+                'eggs': 'spam and eggs'
+            }
+        },
+        [
+            {
+                'henry': 'the first'
+            },
+            {
+                'jacob' : 'the second'
+            }
+        ]
+    ])
+    >>>
+    [
+            {
+                'spam.eggs' : 'spam and eggs'
+            },
+            {
+                '1.henry' : 'the first'
+            },
+            {
+                '2.jacob' : 'the second'
+            }
+    ]
+    """
+
+    result = []
+
+    for element in configuration:
+        if isinstance(element, list):
+            result = result + intermediate_to_list(list(enumerate(element)))
+
+        elif isinstance(element, tuple):
+            (index, config) = element
+            # update the keys to prefix the index
+            temp = {}
+            for k, v in config.items():
+                # write the new key
+                temp['{i}.{key}'.format(i=index + 1, key=k)] = v
+
+            config = temp
+            # now the keys are updated with the index
+            # reduce to atoms the new config
+            # by recursively calling yourself
+            # on a list containing the new atom
+            result = result + intermediate_to_list([config])
+
+        elif isinstance(element, dict):
+            temp = {}
+            for k, v in element.items():
+                if isinstance(v, string_types) or isinstance(v, int):
+                    pass
+                else:
+                    # reduce to atom list
+                    # as v could be dict or list
+                    # enclose it in a flattened list
+                    for son in intermediate_to_list(flatten([v])):
+
+                        for sk, sv in son.items():
+                            nested_key = '{key}.{subkey}'.format(key=k, subkey=sk)
+                            temp[nested_key] = sv
+
+            # now it is atomic, append it to
+            result.append(temp)
+
+        else:
+            raise Exception('malformed intermediate representation')
```

Comment:
It's available as a control for the converters output, it's still easy to mess with things and create something that doesn't make sense